File: techshop-api/src/main/java/com/example/techshop_api/entity/order/Order.java
package com.example.techshop_api.entity.order;

import com.example.techshop_api.entity.user.User;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    User user;

    String orderName;

    String orderAddress;

    String orderEmail;

    String orderPhoneNumber;

    LocalDateTime orderTime;

    String status;

    double totalAmount;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    List<OrderItem> orderItemList;
}

File: techshop-app/src/types/order.ts
export interface Order {
    id: number,
    orderName: string,
    orderAddress: string,
    orderEmail: string,
    orderPhoneNumber: string,
    orderTime: string,
    status: string,
    totalAmount: number
}

export interface OrderDetail {
    id: number,
    userId: number,
    orderName: string,
    orderAddress: string,
    orderEmail: string,
    orderPhoneNumber: string,
    orderTime: string,
    status: string,
    totalAmount: number,
    orderItemList: OrderItemDetail[],
    payment: PaymentDetail
}

export interface OrderItemDetail {
    id: number,
    productName: string,
    sku: string,
    imgUrl: string,
    unitPrice: number,
    quantity: number
}

export interface PaymentDetail {
    id: number,
    amount: number,
    paymentMethod: string,
    paymentStatus: string,
    paymentGateway: string
}